identity-apps-core/apps/authentication-portal/src/main/webapp/includes/branding-preferences.jsp

```diff
@@ -466,6 +486,66 @@
 
             // Proceed only if the branding is enabled.
             if (isBrandingEnabledInTenantPreferences) {
+
+                // Get the resolution strategy.
+                if (brandingPreference.has(THEME_KEY)) {
+                    if (brandingPreference.getJSONObject(THEME_KEY).has(RESOLUTION_STRATEGY_KEY)) {
+                        resolutionStrategy = brandingPreference.getJSONObject(THEME_KEY).getString(RESOLUTION_STRATEGY_KEY);
+                    }
+                }
+
+                // If the branding theme is set using the application.
+                if ("APP_PREFERENCE".equalsIgnoreCase(resolutionStrategy)) {
+
+                    // Check theme from the query param first.
+                    String resolvedUiTheme = request.getParameter("ui_theme");
```

Comment:
cannot we reuse the `uiThemeParam` variable here?

Reply:
We keep resolvedUiTheme separate so we can apply the resolution order.